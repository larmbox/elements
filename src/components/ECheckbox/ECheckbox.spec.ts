import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { ECheckbox } from '~/index';

test('mount component', async () => {
  expect(ECheckbox).toBeTruthy();

  const component = mount(ECheckbox, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain(
    'EFormControl ECheckbox ECheckbox--md ECheckbox--primary'
  );
  expect(component.html()).not.toContain('EFormControl--field');
});

test('register css variables', async () => {
  mount(ECheckbox);
  expect(
    document.getElementById('EThemeProvider--global')!.innerText
  ).toContain('ECheckbox');
});

test('inherit attributes', async () => {
  const component = mount(ECheckbox, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
