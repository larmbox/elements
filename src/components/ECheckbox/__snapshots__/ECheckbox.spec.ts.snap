// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"EFormControl ECheckbox ECheckbox--md ECheckbox--primary\\">
  <div class=\\"ECheckbox-inner\\">
    <div><input id=\\"id\\" type=\\"checkbox\\" />
      <div class=\\"ECheckbox-box\\"><span class=\\"EIcon check ECheckbox-icon\\" aria-hidden=\\"true\\"></span></div>
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;
